lib/routes/objects.js

```diff
@@ -367,6 +367,67 @@ module.exports =
         }
     });
 
+    fastify.route({
+      method: 'GET',
+      url: '/objects/:name/versions',
+      schema: {
+        params: {
+          type: 'object',
+          properties: {
+            name: { type: 'string' }
+          }
+        },
+        response: {
+          200: {
+            type: 'array',
+            items: {
+              type: 'object',
+              properties: {
+                version: { type: 'string' },
+                environments: { type: 'array', items: { type: 'string' } }
+              }
+            }
+          }
+        }
+      },
+      preHandler: fastify.auth([fastify.verifyAuthentication]),
+      handler:
+        /**
+         * @type {FastifyHandler}
+         */
+        async (req, res) => {
+          const {
+            params: { name }
+          } = req;
+
+          const envs = await fastify.getEnvs({ name });
+          const versionsByEnv = await Promise.all(
```

Comment:
`Promise.all` will reject as soon as one of the Promises in the array rejects.
`Promise.allSettled` will never reject, it will resolve once all Promises in the array have either rejected or resolved.

What race conditions are you referring to?

Reply:
That's fair i was thinking you would allow the other env's to complete vs. failing them all. Which is why I suggested `allSettled` that way you can provide to the end user which env failed.